akka-actor-typed/src/main/scala/akka/actor/typed/receptionist/Receptionist.scala

```diff
@@ -25,25 +20,20 @@ class Receptionist(system: ActorSystem[_]) extends Extension {
   }
 
   val ref: ActorRef[Receptionist.Command] = {
-    val behavior =
-      if (hasCluster)
+    val provider: ReceptionistBehaviorProvider =
+      if (hasCluster) {
         system.dynamicAccess
           .createInstanceFor[ReceptionistBehaviorProvider]("akka.cluster.typed.internal.receptionist.ClusterReceptionist$", Nil)
           .recover {
             case ex ⇒
-              system.log.error(
-                ex,
-                "ClusterReceptionist could not be loaded dynamically. Make sure you have all required binaries on the classpath.")
-              ReceptionistImpl
-          }.get.behavior
-
-      else ReceptionistImpl.localOnlyBehavior
+              throw new RuntimeException("ClusterReceptionist could not be loaded dynamically. Make sure you have all required binaries on the classpath.")
```

Comment:
shall we add "(e.g. `akka-cluster`)"?

Reply:
👍 Good idea. I actually ran into that myself yesterday looking into another bug